src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinHttpWebSocket.cs

```diff
@@ -441,6 +448,12 @@ private Task<bool> InternalReceiveAsync(ArraySegment<byte> buffer)
                 Abort();
             }
 
+            // If the task has completed, there's no need to force continuations to run async
+            if (_operation.TcsReceive.Task.Status == TaskStatus.RanToCompletion)
+            {
+                return Task.FromResult(_operation.TcsReceive.Task.GetAwaiter().GetResult());
+            }
```

Comment:
Ah, okay I understand now. WCF doesn't use ContinueWith, but as one of the concerns for fixing this bug was a worry about performance degedation, the if block would avoid jumping threads if someone is using ContinueWith. A new task is only allocated on the ReceiveAsync code path as SendAsync uses Task.CompletedTask. But thinking about it, the extra allocation for the await case isn't worth it for this one scenario. I'll remove the extra if's.